code/game/machinery/dance_machine.dm

```diff
@@ -453,3 +555,229 @@
+	sharpness = SHARP_EDGED
+	resistance_flags = NONE
+	max_integrity = 45
+	var/datum/track/R = new ()
+
+/obj/item/record_disk/Initialize() //moves them a bit so they are not all accumulated on the same pixel
+	..()
+	name = "Generic record disk" // the name changes with music
+	pixel_x = rand(-3, 3)
+	pixel_y = rand(-3, 3)
+
+/obj/item/record_disk/throw_impact() //since they are not made with petrolium they are a bit more fragile
+	..()
+	src.visible_message("<span class ='warning'>The spinning [src] shatters on impact!</span>")
+	Destroy()
+
+/obj/item/record_disk/Destroy()
+	playsound(src, 'sound/effects/record_shatter.ogg', 100, 0)
+	new /obj/item/record_shard(get_turf(src))
+	new /obj/item/record_shard(get_turf(src))
+	new /obj/item/record_shard(get_turf(src))
+	new /obj/item/record_shard(get_turf(src))
+	new /obj/item/record_shard(get_turf(src))
+	..()
+/obj/item/record_shard //if the disk breaks you get pieces of disk that can cut into your feet if you are not carefull
+	name = "record disk shard"
+	desc = "A broken shard of a record disk. Who knew record disks were so fragile."
+	icon = 'icons/obj/machines/disk_recorder.dmi'
+	icon_state = "record_shard"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	force = 5
+	throwforce = 8
+	attack_verb = list("stabs", "DISKED", "sliced")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	resistance_flags = ACID_PROOF
+	armor = list("melee" = 100, "bullet" = 0, "laser" = 0, "energy" = 100, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 100)
+	max_integrity = 40
+	sharpness = SHARP_POINTY
+
+/obj/item/record_shard/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is slitting [user.p_their()] [pick("wrists", "throat")] with the record disk shard! It looks like [user.p_theyre()] trying to commit suicide.</span>")
+	return (BRUTELOSS)
+
+/obj/item/record_shard/Initialize()
+	. = ..()
+	AddComponent(/datum/component/caltrop, force)
+	AddComponent(/datum/component/butchering, 150, 65)
+	pixel_x = rand(-8, 8)
+	pixel_y = rand(-8, 8)
+
+/obj/item/record_shard/afterattack(atom/A as mob|obj, mob/user, proximity)
+	. = ..()
+	if(!proximity || !(src in user))
+		return
+	if(isturf(A))
+		return
+	if(istype(A, /obj/item/storage))
+		return
+	var/hit_hand = ((user.active_hand_index % 2 == 0) ? "r_" : "l_") + "arm"
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(!H.gloves && !HAS_TRAIT(H, TRAIT_PIERCEIMMUNE)) // golems, etc
+			to_chat(H, span_warning("[src] cuts into your hand!"))
+			H.apply_damage(force*0.5, BRUTE, hit_hand)
+	else if(ismonkey(user))
+		var/mob/living/carbon/monkey/M = user
+		if(!HAS_TRAIT(M, TRAIT_PIERCEIMMUNE))
+			to_chat(M, span_warning("[src] cuts into your hand!"))
+			M.apply_damage(force*0.5, BRUTE, hit_hand)
+
+/obj/machinery/gramophone_recorder //used to record new tracks to add to the jukeboxes adminspawn
```

Comment:
This should not be easily accessible to ANY player, add a dev note that reinforces this

Reply:
done, it now clearly specify  that admin should be careful when handling this piece of technology